src/appleseed/renderer/modeling/aov/cryptomatte.h

```diff
@@ -0,0 +1,133 @@
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Sergo Pogosyan, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#ifndef APPLESEED_RENDERER_MODELING_AOV_CRYPTOMATTEAOV_H
+#define APPLESEED_RENDERER_MODELING_AOV_CRYPTOMATTEAOV_H
+
+// appleseed.renderer headers.
+#include "renderer/modeling/aov/aov.h"
+#include "renderer/modeling/aov/iaovfactory.h"
+
+// appleseed.foundation headers.
+#include "foundation/utility/autoreleaseptr.h"
+
+// BCD headers.
+#include "bcd/DeepImage.h"
+
+// Standard headers.
+#include <cstddef>
+
+// Forward declarations.
+namespace foundation    { class ImageAttributes; }
+namespace renderer      { class AOV; }
+namespace renderer      { class ParamArray; }
+
+namespace renderer
+{
+
+//
+// Cryptomatte AOV.
+//
+
+class CryptomatteAOV
+  : public AOV
+{
+  public:
+    enum class CryptomatteType
+    {
+        ObjectNames,
+        MaterialNames,
+    };
+
+    CryptomatteAOV(const ParamArray& params);
+
+    ~CryptomatteAOV() override;
+
+    const char* get_model() const override;
+
+    size_t get_channel_count() const override;
+
+    const char** get_channel_names() const override;
+
+    bool has_color_data() const override;
+
+    void create_image(
+        const size_t    canvas_width,
+        const size_t    canvas_height,
+        const size_t    tile_width,
+        const size_t    tile_height,
+        ImageStack&     aov_images) override;
+
+    void clear_image() override;
+
+    foundation::auto_release_ptr<AOVAccumulator> create_accumulator() const override;
+
+    bool write_images(
+        const char*                         file_path,
+        const foundation::ImageAttributes&  image_attributes) const override;
+
+  private:
+    friend class CryptomatteAOVFactory;
+
+    struct                      Impl;
```

Comment:
We only require that data members are aligned with other data members. Since there are no others here, you don't need all these spaces.